config/config.go

```diff
@@ -56,6 +56,15 @@ func IsCloudConfig(userdata string) bool {
 	return (header == "#cloud-config")
 }
 
+func IsMultipartMime(userdata string) bool {
+	header := strings.SplitN(userdata, "\n", 2)[0]
```

Comment:
Can we be sure that the content-type is always going to be found in the first header? I think I'd use `net/textproto` package to read the headers: https://pkg.go.dev/net/textproto@go1.20.4#Reader.ReadMIMEHeader

Reply:
Sounds good.